pkg/v5/configmap/spec.go

```diff
@@ -7,9 +7,18 @@ import (
 )
 
 type Interface interface {
+	GetCurrentState(ctx context.Context, configMapConfig ConfigMapConfig) ([]*corev1.ConfigMap, error)
 	GetDesiredState(ctx context.Context, configMapValues ConfigMapValues) ([]*corev1.ConfigMap, error)
 }
 
+// ConfigMapConfig is used by the configmap resources to provide config to
+// calculate the current state.
+type ConfigMapConfig struct {
+	ClusterID      string
+	GuestAPIDomain string
+	Namespaces     []string
+}
```

Comment:
This struct provides the config for getting the current state. It can add provider specific namespaces if necessary.